remoting/connection.go

```diff
@@ -131,20 +138,58 @@ func (c *clientConnection) handleMessage(msgType messageType, msg []byte) error
 	return nil
 }
 
-func createNetConnection(serverAddress string) (net.Conn, error) {
+func createNetConnection(serverAddress string, tlsConfig conf.TLSConfig) (net.Conn, error) {
 	addr, err := net.ResolveTCPAddr("tcp", serverAddress)
 	if err != nil {
 		return nil, err
 	}
-	nc, err := net.DialTCP("tcp", nil, addr)
+	var nc net.Conn
+	if tlsConfig.Enabled {
+		tlsConf, err := getClientTLSConfig(tlsConfig)
```

Comment:
We should create the *tls.Config once when the Client is created, not every time when we create a connection (we create connections a lot)

Reply:
Ah, good catch. Fixed